Models/Connection/MySqlConnectionInfo.js

```diff
@@ -0,0 +1,173 @@
+import mySql from "mysql2";
+import ConnectionInfo from "./ConnectionInfo.js";
+import MySqlSettingData from "./MySqlSettingData.js";
+import DataSourceCollection from "../../renderEngine/Source/DataSourceCollection.js";
+import CancellationToken from "../../renderEngine/Cancellation/CancellationToken.js";
+import Request from "../request.js";
+import ExceptionResult from "../../renderEngine/Models/ExceptionResult.js";
+import WebServerException from "../Exceptions/WebServerException.js";
+import MysqlRow from "../mysqlRow.js";
+export default class MySqlConnectionInfo extends ConnectionInfo {
+  /** @type {MySqlSettingData} */
+  settings;
+
+  /**
+   * @param {string} name
+   * @param {MySqlSettingData} settings
+   */
+  constructor(name, settings) {
+    super(name);
+    this.settings = settings;
+  }
+
+  /**
+   * @param {NodeJS.Dict<object|string|number>} parameters
+   * @param {CancellationToken} cancellationToken
+   * @returns {Promise<DataSourceCollection>}
+   */
+  async loadDataAsync(parameters, cancellationToken) {
+    /** @type {mySql.Pool?} */
+    let pool = null;
+    pool = mySql.createPool({
+      host: this.settings.host,
+      user: this.settings.password,
+      database: this.settings.database,
+    });
+    pool.getConnection(async (err, connection) => {
```

Comment:
this approach is not good for we. also you forget to release the connection in the end.

```javascript
pool.getConnection(function(err, conn) {
  // Do something with the connection
  conn.query(/* ... */);
  // Don't forget to release the connection when finished!
  pool.releaseConnection(conn);
});```

please use `pool.query()` and `pool.execute()` instead of get connection manually:

```javascrpt
pool.query("SELECT `field` FROM `table`", function(err, rows, fields) {
  // Connection is automatically released when query resolves
});```

for more information visit [package link](https://www.npmjs.com/package/mysql2)